src/EditorFeatures/CSharpTest/InlineMethod/CSharpInlineMethodTests.cs

```diff
@@ -0,0 +1,645 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.CodeRefactorings.InlineMethod;
+using Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.InlineMethod
+{
+    [Trait(Traits.Feature, Traits.Features.CodeActionsInlineMethod)]
+    public class CSharpInlineMethodTests : AbstractCSharpCodeActionTest
+    {
+        protected override CodeRefactoringProvider CreateCodeRefactoringProvider(Workspace workspace, TestParameters parameters)
+            => ((TestWorkspace)workspace).ExportProvider.GetExportedValue<CSharpInlineMethodRefactoringProvider>();
+
+        [Fact]
+        public Task TestInlineMethodWithSingleStatement()
+            => TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        Ca[||]llee(i, j);
+    }
+
+    private void Callee(int i, int j)
+    {
+        System.Console.WriteLine(i + j);
+    }
+}",
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        System.Console.WriteLine(i + j);
+    }
+
+    private void Callee(int i, int j)
+    {
+        System.Console.WriteLine(i + j);
+    }
+}");
+
+        [Fact]
+        public Task TestExtractArrowExpressionBody()
+            => TestInRegularAndScript1Async(@"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        Ca[||]llee(i, j);
+    }
+
+    private void Callee(int i, int j)
+        => System.Console.WriteLine(i + j);
+}",
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+System.Console.WriteLine(i + j);
+    }
+
+    private void Callee(int i, int j)
+        => System.Console.WriteLine(i + j);
+}");
+
+        [Fact]
+        public Task TestExtractExpressionBody()
+            => TestInRegularAndScript1Async(
+                @"
+public class TestClass
+{
+    private void Caller(int i, int j)
+    {
+        var x = Ca[||]llee(i, j);
+    }
+
+    private int Callee(int i, int j)
+        => i + j;
```

Comment:
do you have tests that validate parenthesizatino.  i.e. `Add(i, j) * 2` should become `(i + j) * 2` not `i + j * 2`

Reply:
That's a good point and I think there is no such check in the logic. 
I'll leave this comment unresolved until I push a commit to handle it.

Reply:
Test cases for the parenthesis have been added